pwiz_tools/Skyline/TestRunnerLib/RunTests.cs

```diff
@@ -271,6 +275,53 @@ private static string GetAssemblyPath(string assembly)
             return Path.Combine(runnerExeDirectory, assembly);
         }
 
+        private void RunTestWithTimeout(TestInfo test, object testObject)
+        {
+            // Signals completion of the test
+            var taskCompletedEvent = new ManualResetEvent(false);
+
+            // Run the test on a thread
+            Thread testThread = new Thread(() =>
```

Comment:
I wonder whether this would cause problems with other tests that do stuff with threads. If you run every test with a timeout but make it default to a large number (so they all run in a thread like this), do they all work?

Reply:
Yeah, something in there is unhappy:
![image](https://github.com/ProteoWizard/pwiz/assets/1200761/ab641961-4e8f-4866-8a63-062c900740d7)
I'll try identifying the existing test thread rather than nesting it.

Reply:
(That was with 8 parallel workers, FWIW)

Reply:
I considered that idea of killing the existing test thread but I'm skeptical that you can kill the existing test thread without breaking the RunTests flow. It's worth a try though. I think that DragDrop error probably comes from no longer running on the UI thread and something didn't InvokeRequired/Invoke when it perhaps should be.

Reply:
Obviously MSTest/VSTest is able to do UI tests with a timeout of course, but perhaps it does quite a bit of extra thread voodoo when a timeout is used.